frontend/javascripts/oxalis/view/left-border-tabs/histogram_view.js

```diff
@@ -185,6 +193,80 @@ class Histogram extends React.PureComponent<HistogramProps, HistogramState> {
     500,
   );
 
+  getCuboidForViewport = async (viewport: OrthoView, layerName: string) => {
+    const state = Store.getState();
+
+    const [curX, curY, curZ] = Dimensions.transDim(
+      Dimensions.roundCoordinate(getPosition(state.flycam)),
+      viewport,
+    );
+    const [halfViewportExtentX, halfViewportExtentY] = getHalfViewportExtentsFromState(
+      state,
+      viewport,
+    );
+
+    const min = Dimensions.transDim(
+      V3.sub([curX, curY, curZ], [halfViewportExtentX, halfViewportExtentY, 0]),
+      viewport,
+    );
+    const max = Dimensions.transDim(
+      V3.add([curX, curY, curZ], [halfViewportExtentX, halfViewportExtentY, 1]),
+      viewport,
+    );
+
+    const resolutionIndex = getRequestLogZoomStep(state);
+    const cuboid = await api.data.getDataFor2DBoundingBox(layerName, { min, max }, resolutionIndex);
+    return cuboid;
+  };
+
+  getClippingValues = async (layerName: string, threshold: number = 0.05) => {
+    const cuboidXY = await this.getCuboidForViewport(OrthoViews.PLANE_XY, layerName);
+    const cuboidXZ = await this.getCuboidForViewport(OrthoViews.PLANE_XZ, layerName);
+    const cuboidYZ = await this.getCuboidForViewport(OrthoViews.PLANE_YZ, layerName);
```

Comment:
Promises which don't depend on each other can be parallelized like this:

```suggestion
    const [cuboidXY, cuboidXZ, cuboidYZ] = Promise.all([
      this.getCuboidForViewport(OrthoViews.PLANE_XY, layerName),
      this.getCuboidForViewport(OrthoViews.PLANE_XZ, layerName),
      this.getCuboidForViewport(OrthoViews.PLANE_YZ, layerName),
    ]);
```

Reply:
I get : an index signature declaring the expected key / value type is missing in  `Promise` when I try to implement this.

Reply:
Oh, sorry! :see_no_evil: Glad you got it working :)